operator/apis/loki/v1/lokistack_types.go

```diff
@@ -345,6 +345,58 @@ type ClusterProxy struct {
 	NoProxy string `json:"noProxy,omitempty"`
 }
 
+// HashRingType defines the type of hash ring which can be used with the Loki cluster.
+//
+// +kubebuilder:validation:Enum=memberlist
+type HashRingType string
+
+const (
+	// HashRingMemberList when using memberlist for the distributed hash ring.
+	HashRingMemberList HashRingType = "memberlist"
+)
+
+// MemberListInstanceAddrType defines the type of pod network to use for advertising IPs to the ring.
+type MemberListInstanceAddrType string
```

Comment:
Shorten the type name to `InstanceAddrType`, so that the full name can be used in the constants below?